include/bluetooth/services/dfu_smp_c.h

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef BT_GATT_DFU_SMP_C_H_
+#define BT_GATT_DFU_SMP_C_H_
+
+/**
+ * @file
+ * @defgroup bt_gatt_dfu_smp_c BLE GATT DFU SMP Client API
+ * @{
+ * @brief API for the BLE GATT DFU SMP (DFU_SMP) Client.
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdbool.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/gatt_dm.h>
+
+
+/** @brief SMP Service.
+ *
+ *  The 128-bit service UUID is 8D53DC1D-1DB7-4CD3-868B-8A527460AA84.
+ */
+#define DFU_SMP_UUID_SERVICE BT_UUID_DECLARE_128( \
+				0x84, 0xaa, 0x60, 0x74, 0x52, 0x8a, 0x8b, 0x86,\
+				0xd3, 0x4c, 0xb7, 0x1d, 0x1d, 0xdc, 0x53, 0x8d)
+
+/** @brief SMP Characteristic.
+ *
+ *  The characteristic is used for both requests and responses.
+ *  The UUID is DA2E7828-FBCE-4E01-AE9E-261174997C48.
+ */
+#define DFU_SMP_UUID_CHAR BT_UUID_DECLARE_128( \
+				0x48, 0x7c, 0x99, 0x74, 0x11, 0x26, 0x9e, 0xae,\
+				0x01, 0x4e, 0xce, 0xfb, 0x28, 0x78, 0x2e, 0xda)
+
+
+/* Forward declaration required for function handlers. */
+struct bt_gatt_dfu_smp_c;
+
+/** @brief Header used internally by dfu_smp.
+ */
+struct dfu_smp_header {
+	/** Operation. */
+	u8_t op;
+	/** Operational flags. */
+	u8_t flags;
+	/** Payload length high byte. */
+	u8_t len_h8;
+	/** Payload length low byte. */
+	u8_t len_l8;
+	/** Group ID high byte. */
+	u8_t group_h8;
+	/** Group ID low byte. */
+	u8_t group_l8;
+	/** Sequence number. */
+	u8_t seq;
+	/** Command ID. */
+	u8_t id;
+};
+
+/** @brief Current response state.
+ */
+struct bt_gatt_dfu_smp_rsp_state {
+	/** @brief Total size of the response.
+	 *
+	 *  Total expected size of the response that is updated.
+	 *  This value is retrieved from the first part of the
```

Comment:
Is that correct? I didn't understand the input.

Reply:
Fixed